api/controllers/EventController.js

```diff
@@ -37,9 +37,41 @@ var EventController = {
             })
         })
     },
-    
+
     update: function (req, res) {
-        
+        var eventId = req.swagger.params.id.value;
+        var token = req.headers.authorization;
+
+        auth.verifyToken(token, function (err, userID) {
+            if (err) return res.status(401).json(err.message);
+
+            Event.findById(eventId, function(err, event){
+                if (err) return res.status(500).json(err.message);
+
+                if(userID == event.author){
```

Comment:
did you test this? `userID` is a String while `event.author` is an ObjectId

Reply:
Yeah, === didn't work because their data types are not the same.
So I used == instead.
 -- Tested, works fine

I think we need to cast one of them to make === work.
"The == operator will compare for equality after doing any necessary type conversions. The === operator will not do the conversion, so if two values are not the same type === will simply return false. Both are equally quick."